File: app/src/main/java/com/wuguangxin/simple/mvp/LoginPresenter.java
package com.wuguangxin.simple.mvp;

import com.wuguangxin.mvp.presenter.AbsBasePresenter;
import com.wuguangxin.simple.bean.UserBean;

/**
 * 登录
 * Created by wuguangxin on 17/5/9.
 */
public class LoginPresenter extends AbsBasePresenter<LoginContract.Model, LoginContract.View>
        implements LoginContract.Presenter, LoginContract.Callback {

    public LoginPresenter(LoginContract.View view) {
        super(view);
    }

    @Override
    public LoginContract.Model newModel() {
        return new LoginModel();
    }

    @Override
    public boolean isPull() {
        return true;
    }

    @Override
    public void login(String username, String password) {
        getModel().login(username, password, this);
    }

    @Override
    public void onSuccessLogin(UserBean userBean) {
        getView().onSuccessLogin(userBean);
        getView().onFinish();
    }
}

File: app/src/main/java/com/wuguangxin/simple/mvp/LoginPresenter.kt
package com.wuguangxin.simple.mvp

import com.wuguangxin.mvp.presenter.BasePresenter
import com.wuguangxin.simple.bean.UserBean

/**
 * 登录
 * Created by wuguangxin on 17/5/9.
 */
class LoginPresenter(view: LoginContract.View) :
    BasePresenter<LoginContract.Model, LoginContract.View>(view),
    LoginContract.Presenter,
    LoginContract.Callback {

    override fun newModel(): LoginContract.Model {
        return LoginModel()
    }

    override fun login(username: String, password: String) {
        getModel().login(username, password, this)
    }

    override fun onSuccessLogin(userBean: UserBean?) {
        getView().onSuccessLogin(userBean)
        getView().onFinish()
    }

    override fun isPull(): Boolean {
        return true
    }
}